### DISCUSSION TITLE
Is it possible to add more (boost) relevance to single fields?

### DISCUSSION THREAD

**Discussion by ovenum** [loupe-php/loupe#71] [Q&A] [answered]
Hi 👋
Thanks for all the work being put into this library. Just started using Loupe to add more search functionality to [Kirby Cms](https://getkirby.com). 

This might be a case of _doing it wrong_ but how, for example, can i add more relevance to a search field from a search result?

Adding documents to Loupes search index with the searchable fields: `title` and `text`.  I would like to add more relevance to search results where the title field matches the search query.
So that they rank higher compared to results where the query matches the text field.

**Comment by Toflar** [comment]
This is currently not supported :)

**Reply by ovenum** [reply]
Thanks for your response @Toflar. 
Since Loupe uses TF-IDF to create a documents relevance score for a search query, i could fake boost the title field by just repeating the document title when adding it to loupes search index? 

So for a movie with the title _Star Wars_ adding it to the search with `$loupe->addDocument([ 'title' => 'Star Wars Star Wars Star Wars])`, should give it a higher relevance because it has a higher term frequency.

Documents with matches in the title ranking higher seems to be a common expectation, at least from my experience when discussing search capabilities.

**Reply by Toflar** [reply]
I don't think that would work because repeating the same term would make it **less** important. I think to achieve what you want we would need proper field weighing but not even MeiliSearch supports that. But if you want to give it a try, PRs are very welcome :)

**Reply by ovenum** [reply]
Oh! I have to re-read how TF-IDF works : )

Saw the Feature Request for Meili Search but thought that there must be a different solution then : ) But as you said, the simple trick might not do it. 

I might not be able to do a full PR at the moment and my SQLite knowledge is very limited ( i saw you doing a lot of performance optimisation). Also fully respect if this is currently not planed. 

Was looking into Loupes DB structure and the `CosineSimilarity::fromQuery` function. 
If a term inside the `terms` tracks which field(s) it originates from a relevance calculation per field might be possible by calculating per field relevance scores if it’s requested by the search?

**Reply by Toflar** [reply]
Yeah the relevance sorting is probably the most complex part apart from the typo tolerance.
There is currently no relation between the search term and where it was found when calculating the relevance (although perfectly doable). But even more so, we would first need a concept on how to configure the weighing. I don't want to come up with our own concept but rather see what other search engines did and decide for at least a similar way so developers are potentially familiar with it.

But it's probably going to be pretty time-consuming, not something I could tackle at the moment - even if we had a concept :)

**Reply by ovenum** [reply, marked as answer]
Looks like this will be possible in an upcoming release. 
See #111 for the PR and #113 for the upcoming documentation. Thanks @Toflar and @daun for this awesome work 🎉

Marked this as answer so it’s not shown as open question.
